docs/cpp/customize-cpp-settings.md

```diff
@@ -0,0 +1,175 @@
+
+  A `configurationProvider` candidate extension must implement [vscode-cpptools-api](https://github.com/microsoft/vscode-cpptools-api).
+
+- `mergeConfigurations`
+  Set to `true` to merge include paths, defines, and forced includes with those from a configuration provider.
+
+- `windowsSdkVersion`
+  The versions of the Windows SDK include path to use on Windows, for example `10.0.17134.0`.
+
+- `macFrameworkPath`
+  A list of paths for the IntelliSense engine to use while searching for included headers from Mac frameworks. Only supported on configurations for macOS frameworks.
+
+- `forcedInclude`
+  A list of files that should be included before any text in the source file is processed. Files are included in the order listed.
+
+- `compileCommands`
+  The full path to the `compile_commands.json` file for the workspace. If there is a matching entry in `compile_commands.json` for a file open in the editor, that command line is to configure IntelliSense for that file, instead of the other fields of `c_cpp_properties.json`.
+  For more information about the file format, see the [Clang documentation](https://clang.llvm.org/docs/JSONCompilationDatabase.html). Some build systems, such as CMake, [simplify generating this file](https://cmake.org/cmake/help/v3.5/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html).
+
+- `dotConfig`
+  A path to a `.config` file, created by the Kconfig system. The Kconfig system generates a file with all the defines needed to build a project. Examples of projects that use the Kconfig system are the Linux Kernel and NuttX RTOS.
+
+- `customConfigurationVariables`
+  Custom variables that can be queried through the command `${cpptools:activeConfigCustomVariable}` to use for the input variables in `launch.json` or `tasks.json`.
+
+- `browse`
+  The set of properties used when `"C_Cpp.intelliSenseEngine"` is set to `"Tag Parser"` (also referred to as "fuzzy" IntelliSense, or the "browse" engine). These properties are also used by the **Go to Definition/Declaration** features, or when the "default" IntelliSense engine is unable to resolve the `#includes` in your source files.
+
+- `recursiveIncludes`
+  A set of properties used to configure how the extension processes an `includePath` entry that specifies a recursive search.
+
+### Browse properties
+
+- `path`
+  A list of paths for the Tag Parser to search for headers included by your source files. If omitted, `includePath` is used as the `path`. Searching on these paths is recursive by default. Specify `*` to indicate nonrecursive search. For example, `${workspaceFolder}` searches through all subdirectories while `${workspaceFolder}/*` does not.
```

Comment:
The reference to `Tag Parser` might not be understood by readers. Also, it doesn't just use the "path" to search for headers -- the source files on the path are also tag parsed. Could you reword it to something like "A list of paths that have source files parsed for use in global symbol searches."?

Reply:
Also, in the future, the database may contain non-Tag Parser data, i.e. IntelliSense-based data, so by not referencing "Tag Parser" these docs will be future proof against that potential change.